guides/theming.md

```diff
@@ -1,311 +1,359 @@
+#### Multiple themes across separate files
 
-#### Toggling classes
+You can define multiple themes in seprate files by creating multiple theme files per
+[Defining a theme](#defining-a-theme), adding each of the files to the `styles` of your
+`angular.json`. However, you must additionally set the `inject` option for each of these files to
+`false` in order to prevent all the theme files from being loaded at the same time. When setting
+this property to `false`, your application becomes responsible for manually loading the desired
+file. The approach for this loading depends on your application.
 
-You can use the theming mixins to customize any part of your application with standard
-CSS selectors. For example, let's say you want to toggle alternate colors on a button.
-You would first define a CSS class with the alternate colors.
+### Scoping style customizations
 
-Note that `mat-button-color` should be used instead of `mat-button-theme` as we only
-want to have alternate colors for the button. Using the theme mixin could result in
-duplicative theme styles if the `mat-button-theme` has been included before. Read more about
-this in the [dedicated guide](./duplicate-theming-styles.md).
+You can use Angular Material's Sass mixins to customize component styles within a specific scope
+in your application. The CSS rule declaration which you include a Sass mixin determines its scope.
+The example below shows how to customize the color of all buttons inside elements marked with the
+`.my-special-section` CSS class.
 
 ```scss
-.alternate-button {
-  // Extract the color configuration from the theme and generate
-  // the color theme styles for `mat-button`.
-  @include mat-button-color($alternate-theme);
+@use '~@angular/material' as mat;
+
+.my-special-section {
+ $special-primary: mat.define-palette(mat.$orange-palette);
+ $special-accent: mat.define-palette(mat.$brown-palette);
+ $special-theme: mat.define-dark-theme((
+   color: (primary: $special-primary, accent: $special-accent),
+ ));
+
+ @include mat.button-color($special-theme);
 }
 ```
 
-Then you can use normal Angular class bindings to toggle the alternate styles.
-```html
-<div [class.alternate-button]="isAlternateMode">
-  <button mat-button>Save changes</button>
-</div>
-```
+### Reading hues from palettes
 
-You can use this approach to style any component inside the region marked with the custom
-CSS class.
+You can use the `get-color-from-palette` function to get specific hues from a palette by their
+number identifier. You can also access the contrast color for a particular hue by suffixing the
+hue's number idenfier with `-contrast`.
 
-#### Swapping CSS files
+```scss
+@use '~@angular/material' as mat;
 
-If you want to completely swap a theme without including all the styles at once, you
-can swap the loaded theme file. The details will depend on your application, but the general
-idea looks like this:
+$my-palette: mat.define-palette(mat.$indigo);
```

Comment:
What about an example importing the themes/palettes defined in your main styles definitions, since that'd reduce the number of places you are defining a theme and getting values from it? That's a common use case I've seen people try.

Reply:
I was intending to cover that in a follow-up update to the `theming-your-components` guide. I might move this section there once I start working on that content.